src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/sas_token_auth.py

```diff
@@ -0,0 +1,57 @@
+#---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#---------------------------------------------------------------------------------------------
+
+from base64 import b64encode, b64decode
+from hashlib import sha256
+from hmac import HMAC
+from time import time
+try:
+    from urllib import (urlencode, quote)
+except ImportError:
+    from urllib.parse import (urlencode, quote)  # pylint: disable=import-error,no-name-in-module
+from msrest.authentication import Authentication
+
+
+class SasTokenAuthentication(Authentication):
```

Comment:
>SasTokenAuthentication [](start = 6, length = 22)

This is typically part of the SDK. It seems strange to have it as part of the CLI package. @johanste thoughts?